src/grammar.rs

```diff
@@ -137,7 +142,7 @@ impl Grammar {
     ///     # assert!(sentence_clone.is_ok());
     /// }
     /// ```
-    pub fn generate(&self) -> Result<String, Error> {
+    pub fn generate_seeded(&self, rng: StdRng) -> Result<String, Error> {
```

Comment:
Related to my [previous comment](https://github.com/snewt/bnf/pull/27/files#diff-8733e475042d62732125f511c2b10375R107), does this have to take ownership of the `StdRng` or could it work with a mutable reference?